lib/graphql_schema.rb

```diff
@@ -24,6 +24,11 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def type_from_name(type_name)
+    @types_by_name ||= types.map { |type| [type.name, type] }.to_h
+    @types_by_name[type_name]
```

Comment:
It is usually better to use `.fetch` if we assume then key exists, since it gives a better error message.  Maybe it would be better to just expose the hash and let the caller use the appropriate method to access the type (e.g. `key?`, `[]` or `fetch`).